server/src/main/java/org/opensearch/search/backpressure/settings/SearchShardTaskSettings.java

```diff
@@ -8,23 +8,77 @@
 
 package org.opensearch.search.backpressure.settings;
 
+import org.opensearch.ExceptionsHelper;
 import org.opensearch.common.settings.ClusterSettings;
 import org.opensearch.common.settings.Setting;
 import org.opensearch.common.settings.Settings;
 import org.opensearch.monitor.jvm.JvmStats;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
 /**
  * Defines the settings related to the cancellation of SearchShardTasks.
  *
  * @opensearch.internal
  */
 public class SearchShardTaskSettings {
     private static final long HEAP_SIZE_BYTES = JvmStats.jvmStats().getMem().getHeapMax().getBytes();
+    private final List<Listener> listeners = new ArrayList<>();
 
     private static class Defaults {
+        private static final double CANCELLATION_RATIO = 0.1;
```

Comment:
Could you please add comment what the default value means and why it was chosen, it will probably go to the documentation as well.

Reply:
I have added comments for each setting while defining the setting, in `SearchTaskSettings` and `SearchShardTaskSettings`. About default values, I and @nssuresh2007 have done rounds of tests and observed significant performance improvements with these default values. We will add settings description in documentation as well.